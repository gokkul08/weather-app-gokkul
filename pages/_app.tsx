import React from 'react';
import type { AppProps } from 'next/app';
import { StylesProvider } from '@material-ui/styles';
import { ThemeProvider } from 'styled-components';
import CssBaseline from '@material-ui/core/CssBaseline';
import NavigationBar from '../components/NavigationBar';

const theme = {
  primary: 'green',
}

export default function App({ Component, pageProps}: AppProps):  JSX.Element{

    React.useEffect(() => {
        // Remove the server-side injected CSS.
        const jssStyles = document.querySelector('#jss-server-side');
        if (jssStyles) {
          jssStyles.parentElement.removeChild(jssStyles);
        }
      }, []);

    return (
        <StylesProvider injectFirst>
            <ThemeProvider theme={theme}>
                <CssBaseline />
                <NavigationBar />
                <Component {...pageProps} />
            </ThemeProvider>
        </StylesProvider>
    )
  }