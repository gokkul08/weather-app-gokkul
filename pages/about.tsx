/**
 * This is the about page.
 * This contains information about this application and features
 * This corresponds to /about path
 */

import React from 'react';
import styled from 'styled-components';
import Container from '@material-ui/core/Container';
import Grid from '@material-ui/core/Grid';
import Accordion from '@material-ui/core/Accordion';
import AccordionSummary from '@material-ui/core/AccordionSummary';
import AccordionDetails from '@material-ui/core/AccordionDetails';
import Typography from '@material-ui/core/Typography';
import ExpandMoreIcon from '@material-ui/icons/ExpandMore';

const StyledContainer = styled(Container)`
        padding-top: 20px;
    `;

export const About = () => {
    return (
        <StyledContainer maxWidth="md">
            <Grid container spacing={1}>
                <Grid item xs>
                    <Accordion>
                        <AccordionSummary
                            expandIcon={<ExpandMoreIcon />}
                        >
                        <Typography>What is this website?</Typography>
                        </AccordionSummary>
                        <AccordionDetails>
                        <Typography>
                            This application is created to help users access their weather data
                        </Typography>
                        </AccordionDetails>
                    </Accordion>
                    <Accordion>
                        <AccordionSummary
                            expandIcon={<ExpandMoreIcon />}
                        >
                        <Typography>What features does this website offer?</Typography>
                        </AccordionSummary>
                        <AccordionDetails>
                        <Typography>
                                <li>Users can get Weather data for any location in the world</li>
                                <li>Users can search by City, State and Zip</li>
                                <li>Users can convert weather temperature from °C to °F and vice-versa</li>
                                <li>Users can access a detailed weather page to get more details on their search</li>
                        </Typography>
                        </AccordionDetails>
                    </Accordion>
                    <Accordion>
                        <AccordionSummary
                            expandIcon={<ExpandMoreIcon />}
                        >
                        <Typography>What technologies does this website use?</Typography>
                        </AccordionSummary>
                        <AccordionDetails>
                        <Typography>
                           This website uses NextJS, ReactJS, TypeScript, Styled Components and Material UI
                        </Typography>
                        </AccordionDetails>
                    </Accordion>
                    <Accordion>
                        <AccordionSummary
                            expandIcon={<ExpandMoreIcon />}
                        >
                        <Typography>Author</Typography>
                        </AccordionSummary>
                        <AccordionDetails>
                        <Typography>
                            This website is developed by Annamalai Gokkul Natarajan. I can be reached at agn253@nyu.edu
                        </Typography>
                        </AccordionDetails>
                    </Accordion>
                </Grid>
            </Grid>
        </StyledContainer>
    );
};

export default About;