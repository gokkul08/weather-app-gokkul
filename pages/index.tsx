import Container from '@material-ui/core/Container';
import WeatherContainer from '../components/WeatherComponents/WeatherContainer';

export const Home = () => {
  return (
  <div>
    <Container maxWidth="md">
      <WeatherContainer />
    </Container>
  </div>
  )
};

export default Home;
