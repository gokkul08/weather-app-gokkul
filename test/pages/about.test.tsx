import React from 'react'
import { render } from '../test-utils'
import { About } from '../../pages/about'

describe('About page', () => {
  it('matches snapshot', () => {
    const { asFragment } = render(<About />, {})
    expect(asFragment()).toMatchSnapshot()
  })
})