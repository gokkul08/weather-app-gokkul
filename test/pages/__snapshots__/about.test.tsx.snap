// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About page matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiContainer-root about__StyledContainer-e7e2cb-0 kWwa-DU MuiContainer-maxWidthMd"
  >
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1"
    >
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-true"
      >
        <div
          class="MuiPaper-root MuiAccordion-root MuiAccordion-rounded MuiPaper-elevation1 MuiPaper-rounded"
        >
          <div
            aria-disabled="false"
            aria-expanded="false"
            class="MuiButtonBase-root MuiAccordionSummary-root"
            role="button"
            tabindex="0"
          >
            <div
              class="MuiAccordionSummary-content"
            >
              <p
                class="MuiTypography-root MuiTypography-body1"
              >
                What is this website?
              </p>
            </div>
            <div
              aria-disabled="false"
              aria-hidden="true"
              class="MuiButtonBase-root MuiIconButton-root MuiAccordionSummary-expandIcon MuiIconButton-edgeEnd"
            >
              <span
                class="MuiIconButton-label"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
                  />
                </svg>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </div>
          </div>
          <div
            class="MuiCollapse-container MuiCollapse-hidden"
            style="min-height: 0px;"
          >
            <div
              class="MuiCollapse-wrapper"
            >
              <div
                class="MuiCollapse-wrapperInner"
              >
                <div
                  role="region"
                >
                  <div
                    class="MuiAccordionDetails-root"
                  >
                    <p
                      class="MuiTypography-root MuiTypography-body1"
                    >
                      This application is created to help users access their weather data
                    </p>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="MuiPaper-root MuiAccordion-root MuiAccordion-rounded MuiPaper-elevation1 MuiPaper-rounded"
        >
          <div
            aria-disabled="false"
            aria-expanded="false"
            class="MuiButtonBase-root MuiAccordionSummary-root"
            role="button"
            tabindex="0"
          >
            <div
              class="MuiAccordionSummary-content"
            >
              <p
                class="MuiTypography-root MuiTypography-body1"
              >
                What features does this website offer?
              </p>
            </div>
            <div
              aria-disabled="false"
              aria-hidden="true"
              class="MuiButtonBase-root MuiIconButton-root MuiAccordionSummary-expandIcon MuiIconButton-edgeEnd"
            >
              <span
                class="MuiIconButton-label"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
                  />
                </svg>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </div>
          </div>
          <div
            class="MuiCollapse-container MuiCollapse-hidden"
            style="min-height: 0px;"
          >
            <div
              class="MuiCollapse-wrapper"
            >
              <div
                class="MuiCollapse-wrapperInner"
              >
                <div
                  role="region"
                >
                  <div
                    class="MuiAccordionDetails-root"
                  >
                    <p
                      class="MuiTypography-root MuiTypography-body1"
                    />
                    <li>
                      Users can get Weather data for any location in the world
                    </li>
                    <li>
                      Users can search by City, State and Zip
                    </li>
                    <li>
                      Users can convert weather temperature from °C to °F and vice-versa
                    </li>
                    <li>
                      Users can access a detailed weather page to get more details on their search
                    </li>
                    <p />
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="MuiPaper-root MuiAccordion-root MuiAccordion-rounded MuiPaper-elevation1 MuiPaper-rounded"
        >
          <div
            aria-disabled="false"
            aria-expanded="false"
            class="MuiButtonBase-root MuiAccordionSummary-root"
            role="button"
            tabindex="0"
          >
            <div
              class="MuiAccordionSummary-content"
            >
              <p
                class="MuiTypography-root MuiTypography-body1"
              >
                What technologies does this website use?
              </p>
            </div>
            <div
              aria-disabled="false"
              aria-hidden="true"
              class="MuiButtonBase-root MuiIconButton-root MuiAccordionSummary-expandIcon MuiIconButton-edgeEnd"
            >
              <span
                class="MuiIconButton-label"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
                  />
                </svg>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </div>
          </div>
          <div
            class="MuiCollapse-container MuiCollapse-hidden"
            style="min-height: 0px;"
          >
            <div
              class="MuiCollapse-wrapper"
            >
              <div
                class="MuiCollapse-wrapperInner"
              >
                <div
                  role="region"
                >
                  <div
                    class="MuiAccordionDetails-root"
                  >
                    <p
                      class="MuiTypography-root MuiTypography-body1"
                    >
                      This website uses NextJS, ReactJS, TypeScript, Styled Components and Material UI
                    </p>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="MuiPaper-root MuiAccordion-root MuiAccordion-rounded MuiPaper-elevation1 MuiPaper-rounded"
        >
          <div
            aria-disabled="false"
            aria-expanded="false"
            class="MuiButtonBase-root MuiAccordionSummary-root"
            role="button"
            tabindex="0"
          >
            <div
              class="MuiAccordionSummary-content"
            >
              <p
                class="MuiTypography-root MuiTypography-body1"
              >
                Author
              </p>
            </div>
            <div
              aria-disabled="false"
              aria-hidden="true"
              class="MuiButtonBase-root MuiIconButton-root MuiAccordionSummary-expandIcon MuiIconButton-edgeEnd"
            >
              <span
                class="MuiIconButton-label"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
                  />
                </svg>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </div>
          </div>
          <div
            class="MuiCollapse-container MuiCollapse-hidden"
            style="min-height: 0px;"
          >
            <div
              class="MuiCollapse-wrapper"
            >
              <div
                class="MuiCollapse-wrapperInner"
              >
                <div
                  role="region"
                >
                  <div
                    class="MuiAccordionDetails-root"
                  >
                    <p
                      class="MuiTypography-root MuiTypography-body1"
                    >
                      This website is developed by Annamalai Gokkul Natarajan. I can be reached at agn253@nyu.edu
                    </p>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
